packages/query/src/block-processor.ts

```diff
@@ -189,7 +193,13 @@ export class BlockProcessor implements BlockProcessorInterface {
         throw new Error(`Unexpected block height: ${compactBlock.height} at ${currentHeight}`);
       }
 
-      await this.processBlock(compactBlock, latestKnownBlockHeight);
+      // Set the skip_trial_decrypt flag
+      const skipTrialDecrypt = Boolean(
+        this.walletCreationBlockHeight && currentHeight < BigInt(this.walletCreationBlockHeight),
```

Comment:
skips trial decryption for fresh wallets until wallet creation height

Reply:
<img width="985" alt="Screenshot 2024-08-28 at 8 03 02 PM 2" src="https://github.com/user-attachments/assets/7445053f-8cf4-4781-9fc7-e34068fe90f7">

Reply:
If `walletCreationBlockHeight` is set to zero, the wallet will need to sync the entire chain and trial decrypt every compact block. This occurs when the user does not specify a wallet birthday height during the import of an existing seed phrase. From manual testing, this behavior works correctly when the field is set to zero, but I'll still write some unit tests for completeness.

Reply:
![Uploading Screenshot 2024-08-30 at 10.30.59 PM.png…]()

Reply:
<img width="976" alt="Screenshot 2024-08-30 at 10 30 59 PM" src="https://github.com/user-attachments/assets/b62d93fa-af17-4257-bdb2-2799f2e73222">